### PR TITLE
Add a banner to the build file

### PR DESCRIPTION
**Pull request by paulfalgout**
A bit of a noob at gulp.  This works, but the gulp might needs some adjusting.

Essentially this just adds
```js
/**
 * babel-library-boilerplate - Author libraries in ES6 for Node and the browser..
 * @version v3.0.0
 * @link https://github.com/babel/babel-library-boilerplate.git
 * @license MIT
 */
```

For whatever settings you have in the `package.json`

_Simplified to use esperanto's banner_

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ const esperanto = require('esperanto');
 const browserify = require('browserify');
 const runSequence = require('run-sequence');
 const source = require('vinyl-source-stream');
+const header = require('gulp-header');
```
Comment (jmeas) [inline gulpfile.js]: Remove this line

**Comment by jmeas** [comment]
I'm gonna think about this one before merging. Idk if I want to support banners outta tha box. If not, it would at least go in the Wiki.

File: gulpfile.js (hunk 1 above)
Comment (paulfalgout) [inline gulpfile.js]: Ah yeah.. oops.. Had a different implementation.. then realized esperanto does it.

**Comment by paulfalgout** [comment]
Yep.  Sounds good

File: gulpfile.js [hunk 2]
```diff
@@ -64,13 +64,28 @@ gulp.task('lint-test', function() {
...
+function getBanner() {
+  var banner = ['/**',
+    ' * <%= pkg.name %> - <%= pkg.description %>',
+    ' * @version v<%= pkg.version %>',
+    ' * @link <%= pkg.homepage %>',
+    ' * @license <%= pkg.license %>',
+    ' */',
+    ''].join('\n');
+
+  return _.template(banner)({pkg: manifest});
```
Comment (jmeas) [inline gulpfile.js]: Does this work w.o. requiring in `_` at the top?

Comment (jmeas) [inline gulpfile.js]: Also, could you not just pass in `manifest` and access the data directly?

```js
_.template(banner)({manifest})
```

```js
@license <%= license %>
```

Comment (paulfalgout) [inline gulpfile.js]: It didn't work without adding lodash.

true.. this was a result of a lame copy-pasta from that gulp-header library I was playing with :stuck_out_tongue_winking_eye:

Comment (jmeas) [inline gulpfile.js]: Ah heh :stuck_out_tongue:

**Comment by jmeas** [comment]
Thanks @paulfalgout! Nice work here.

File: gulpfile.js [hunk 3]
```diff
@@ -64,13 +65,28 @@ gulp.task('lint-test', function() {
...
 // Build two versions of the library
 gulp.task('build', ['lint-src', 'clean'], function(done) {
+  var banner = getBanner();
+
   esperanto.bundle({
     base: 'src',
     entry: config.entryFileName,
   }).then(function(bundle) {
     var res = bundle.toUmd({
+      banner: banner,
```
Comment (jmeas) [inline gulpfile.js]: `banner: getBanner(),`

**Comment by jmeas** [comment]
Gonna close this for now. Maybe if it's easier to configure when this is a yeoman we can revisit adding it. Feel free to make a wiki page about this @paulfalgout!

**Pull request closed without merging**